io/io/test/TFileTests.cxx

```diff
@@ -130,6 +130,7 @@ void TestReadWithoutGlobalRegistrationIfPossible(const char *fname)
 // https://github.com/root-project/root/issues/10742
 #ifndef R__WIN32
 // We prefer not to read remotely files from Windows, if possible
+#ifdef R__HAS_DAVIX
```

Comment:
This is effective.  But do we understand why the test fail?  For example, is there a missing feature in the `TWebFile` or are the test "really" needing `Davix`?

Reply:
`TWebFile` does not support HTTP protocol 2.0, only 1.1.

See my comment here:

https://github.com/root-project/root/issues/8890#issuecomment-911515928

Reply:
This may or may not be the root cause here ... The error message I see:
```
Error in <TWebFile::TWebFile>: http://root.cern.ch/files/na49.root? does not exist
```
where the trailing interrogation mark is suspicious.

Reply:
The interrogation mark comes from:

```
fBasicUrl += "?";
fBasicUrl += fUrl.GetOptions();
```

I have added the following guard:

```
      if (strlen(fUrl.GetOptions())) {
         fBasicUrl += "?";
         fBasicUrl += fUrl.GetOptions();
      }
```

and now I get with `TWebFile f{"https://root.cern/files/na49.root"};`:

`Error in <TWebFile::TWebFile>: http://root.cern:443/files/na49.root does not exist`

without the question mark.

The error stems from the ::GetHead function. When calling with message:

`HEAD https://root.cern:443/files/na49.root HTTP/1.0\r\nUser-Agent: ROOT-TWebFile/1.1\r\n\r\n`

one gets as return string: `HTTP/1.0 404 Not Found`

and the 404 return code is forwarded as "-2" to the constructor.

Reply:
Thanks.